yourbench/utils/dataset_engine.py

```
@@ -8,342 +10,273 @@
-            if subset and isinstance(dataset, DatasetDict):
-                if subset in dataset:
-                    return dataset[subset]
-                else:
-                    logger.warning(f"Subset '{subset}' not found in local dataset. Returning empty dataset.")
-                    return Dataset.from_dict({})
-            return dataset
-        else:
-            logger.warning(f"local_dataset_dir '{local_dataset_dir}' does not exist.")
-            if _is_offline_mode():
-                raise ValueError("Offline mode is enabled but local dataset not found")
-            else:
-                logger.warning("Falling back to Hugging Face Hub.")
 
-    # If we're in offline mode and made it here, the local dataset doesn't exist
+def _validate_repo_id(settings: _HFSettings) -> None:
+    """Ensure the constructed repo name is syntactically valid using the Hub API."""
+
     if _is_offline_mode():
-        logger.warning("Offline mode enabled but no local dataset found. Returning empty dataset.")
+        return
+
+    try:
+        HfApi().repo_info(repo_id=settings.repo_id, repo_type="dataset", token=settings.token)
+    except HFValidationError as exc:
+        raise ConfigurationError(f"Invalid repo id '{settings.repo_id}': {exc}") from exc
+    except Exception:
+        pass
+
+
+# ---------------------------------------------------------------------------
+# Local filesystem helpers
+# ---------------------------------------------------------------------------
+
+
+def _load_from_local(path: Path, subset: Optional[str]) -> Dataset:
+    """Return the dataset (or split) located at *path*."""
+
+    ds = load_from_disk(str(path))
+
+    if subset is None or not isinstance(ds, DatasetDict):
+        return ds
+
+    if subset in ds:
+        return ds[subset]
+
+    logger.warning(f"Subset '{subset}' not found in local dataset; returning empty Dataset")
+    return Dataset.from_dict({})
```

Instead of returning an empty dataset silently, raise an error:
```
raise ConfigurationError(f"Subset '{subset}' not found in local dataset")
```
This prevents silent failures downstream

---

Changed to raise `ConfigurationError(f"Subset '{subset}' not found in local dataset")`.